rdtools/analysis_chains.py

```diff
@@ -464,6 +464,18 @@ def _call_clearsky_filter(filter_string):
             f = filtering.clip_filter(
                 self.pv_power, **self.filter_params['clip_filter'])
             filter_components['clip_filter'] = f
+        if 'hour_angle_filter' in self.filter_params:
+            if self.pvlib_location is None:
```

Comment:
if undefined, self.pvlib_location won't be None, it'll actually return an AttributeError.  I've got a fix for the handful of places this crops up.